src/common/snippets/src/pass/mha_tokenization.cpp

```diff
@@ -224,20 +234,14 @@ ov::snippets::pass::TokenizeMHASnippets::TokenizeMHASnippets(const SnippetsToken
         // Example:
         //     Buffer - i32 [32, 128] -> ~ Loop ~ -> Buffer - i8 [32, 128]
         //     After each Loop iteration we should increment pointers of Buffers: accordingly on 4 byte and 1 byte for scalar case.
-        //     It means that these Buffers cannot be inplace => Each Buffer should have the own register
+        //     It means that these increments are not proportional => Each Buffer should have the own register
         // For that we can just check the following "branches":
         //  - Between MatMul0 and MatMul1 - Softmax is sync point. The operations between MatMul0 -> Softmax and Softmax -> MatMul1
         //                                  will be fused into one loop after conversion to snippet dialect (Because it's just FQ, Eltwise nodes)
-        //  - Between MatMul0 and Transpose1 - At the moment operations after Transpose1 cannot be fused in Transpose Loop (to avoid performance regressions).
+        //  - Between MatMul0 and Transpose1 - At the moment operations after Transpose1 cannot be fused in inner Transpose Loop
+        //                                     (to avoid performance regressions due to scalar calculations).
         //                                     But operations after Transpose1 and before MatMul0  will be fused into one loop as well (look at first point)
-        // Note: If the pass is updated, need to check the new possible branches for potential non-inplace Buffers!
-        // Default value is 2 because
-        //  - Firstly, Softmax always needs Buffers
-        //  - Secondly, Softmax needs 2 Buffers but they can be inplace - One virtual port is enough for Softmax => buffer_count = 1
-        //  - Thirdly, MatMul requires unique Buffers on inputs and outputs because blocking implementation increments input/output pointers during computations
-        //    However, all of the Buffers are usually reused by the next MatMul and Softmax.
-        //    So on sufficiently large subgraphs we use only one additional unique buffer => buffer_count increments by 1
-        size_t buffer_count = 2;
+        size_t uniqie_buffer_reg_group_count = 1;  // After MatMul0 there is always one Buffer
```

Comment:
Why we previously needed 2 buffers, but now only one?
Is it because of the blocking support in dynamism?

Reply:
Because this part was outdated - it was written before feature about Buffer optimization.
Now the default value is one since we always need to have one buffer at least.
For example, if we take unet model with MatMul->Softmax->MatMul, there will be 3 Buffer with the same Reg Group -> `uniqie_buffer_reg_group_count = 1`